src/server/handlers.rs

```diff
@@ -762,3 +747,158 @@ impl<P: Publisher> Handler<P> {
+                    component.name
+                ));
+            }
+
+            if !component_details.insert(image_name.to_string()) {
+                return Some(format!(
+                    "Duplicate image reference in manifest: {}",
+                    image_name
+                ));
+            }
+
+            if !link_name_set.insert(link.to_string()) {
+                return Some(format!("Duplicate link name in manifest: {}", image_name));
+            }
+        }
+
+        // Actor validation : Actors should have a unique name and reference
+        if let Properties::Actor {
+            properties: ActorProperties { image: image_name },
+        } = &component.properties
+        {
+            if !component_details.insert(image_name.to_string()) {
+                return Some(format!(
+                    "Duplicate image reference in manifest: {}",
+                    image_name
+                ));
+            }
+        }
+
+        // Linkdef validation : A linkdef from a component should have a unique target and reference
+        let mut linkdef_set: HashSet<String> = HashSet::new();
+        if let Some(traits_vec) = &component.traits {
+            for trait_item in traits_vec.iter() {
+                if let Trait {
+                    // TODO : add trait type validation after custom types are done. See TraitProperty enum.
+                    properties:
+                        TraitProperty::Linkdef(LinkdefProperty {
+                            target: target_name,
+                            values: linkdef_values,
+                        }),
+                    ..
+                } = &trait_item
+                {
+                    if !linkdef_set.insert(target_name.to_string()) {
+                        return Some(format!(
+                            "Duplicate target for linkdef in manifest: {}",
+                            target_name
+                        ));
+                    }
+
+                    if !existing_linkdefs.is_empty()
+                        && linkdef_values.is_some()
+                        && !(existing_linkdefs
+                            .get(&(component.name.clone(), target_name.to_string()))
+                            .map(|x| x.to_owned())
+                            .eq(linkdef_values)
+                            || existing_linkdefs
+                                .get(&(target_name.to_string(), component.name.clone()))
+                                .map(|x| x.to_owned())
+                                .eq(linkdef_values))
+                    {
+                        return Some(format!(
+                            "Component {} and target {} have different linkdef values from earlier version of manifest",
+                            component.name, target_name
+                        ));
+                    }
+                }
+            }
+        }
+    }
+    None
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I'd love to see some example manifests that fail the validation based on the different conditions above